### PR TITLE
Probability Initial Page

### PR DESCRIPTION
**Pull request by amirhmk**
This PR is the initial work to create the `probability` experience. It adds the following functionality: 

1. Dice component
2. Card component (For writing text on a plane)
3.  [Physics Engine](https://www.google.com/search?client=safari&rls=en&q=a+frame+physics+engine&ie=UTF-8&oe=UTF-8)

I think we can do a little better on tidying up the directory and keeping it clean as a part of this PR. 

Dices that are being thrown from the sky start at a random position (where you can see) with a random rotation, which should give us what we need for our experiment.

### CODE REVIEW

File: src/components/probability/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      type: "number",
+      default: 0,
+    },
+  },
+  init: function () {
+    const probability = this;
+    // Card for results
+    const resultsEl = document.createElement("a-entity");
+    resultsEl.setAttribute("card", {
+      text: "Results\n\n1: 0\n2: 0\n3: 0\n4: 0\n5: 0\n6: 0",
```
Comment (amirhmk) [inline src/components/probability/index.js:13]: I think this bit can be refactored into a function

File: src/scene.html [hunk 2]
```diff
@@ -1,44 +1,60 @@
...
+        color="blue"
+        static-body
+        width="1000"
+        height="1"
+        depth="1000"
+        visible="false"
+      ></a-box>
+      <a-entity
+        card="pos_x: -4.5; pos_y: 0.3; rot_x: -15; rot_y:20; text:Results\n\n1: 0\n2: 2\n3: 0\n4: 5\n6: 7"
+        class="probability" class="probability" visible=false
```
Comment (quentinv72) [inline src/scene.html:41]: Remove a class

File: src/scene.html [hunk 3]
```diff
@@ -1,44 +1,60 @@
...
+        id="one"
+        position="2 -2.5 -3"
+        color="blue"
+        static-body
+        width="1000"
+        height="1"
+        depth="1000"
+        visible="false"
+      ></a-box>
+      <a-entity
```
Comment (quentinv72) [inline src/scene.html:39]: I think this can be removed

File: src/components/title.js [hunk 4]
```diff
@@ -23,6 +23,12 @@ AFRAME.registerComponent("title", {
       y: 6,
       z: -8,
     });
+    const el = this.el;
```
Comment (quentinv72) [inline src/components/title.js:26]: This can be removed

File: src/components/title.js [hunk 5]
```diff
@@ -23,6 +23,12 @@ AFRAME.registerComponent("title", {
...
+    const sceneEl = el.sceneEl;
+    // eslint-disable-next-line no-unused-vars
+    sceneEl.addEventListener("remove", function (event) {
+      sceneEl.remove(el);
```
Comment (quentinv72) [inline src/components/title.js:30]: This can be removed

File: src/components/probability/index.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
+AFRAME.registerComponent("probability", {
+  schema: {
+    isActive: {
+      type: "number",
+      default: 0,
+    },
+  },
+  init: function () {
+    const probability = this;
```
Comment (quentinv72) [inline src/components/probability/index.js:9]: What is this line for?

File: src/components/probability/index.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    if (this.data.isActive) {
+      const allElements = document.querySelectorAll(" .cardProb");
+      allElements.forEach((el) =>
+        el.setAttribute("animation", {
+          property: "scale",
+          to: "1 1 1",
+          dur: 1000,
+        })
+      );
+    }
```
Comment (quentinv72) [inline src/components/probability/index.js:89]: Why not do this directly in init or add animation component in card.js?

File: src/components/welcomeScreen.js [hunk 8]
```diff
@@ -40,9 +40,13 @@ AFRAME.registerComponent("welcome", {
...
       button.setAttribute("geometry", { width: 1.5, height: 0.5 });
     });
-
     button.addEventListener("click", function () {
       button.emit("fadeaway", { className: "welcome-page" }, true);
+      setTimeout(() => {
+        document
+          .querySelector("a-scene")
+          .setAttribute("probability", { isActive: 1 });
+      }, 1200);
```
Comment (quentinv72) [inline src/components/welcomeScreen.js:49]: We may need to add this directly to fadeaway component

**Comment by quentinv72** [comment]
I'm not entirely convinced by the idea of writing one element tag `<a-entity probability>` with component probability to build entire probability scene... I think we can have it broken down in the cene.html, it might be easier to read. What do you think?

**Pull request merged**